### PR TITLE
[2017.7] event send multi master

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
When using Salt multi-master, if we encounter a salt master that has not accepted the minion key yet we should not exit right away, rather continue on and try the next salt master available in the list.  If the channel.send is successful, ensure that `ret` is True in case previous attempts at contacting masters had set it to False.

### What issues does this PR fix or reference?
#48415 

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by rallytime — APPROVED** [review]
This is nice.

File: salt/crypt.py [hunk 1]
```diff
@@ -520,9 +520,17 @@ def _authenticate(self):
                     error = SaltClientError('Detect mode is on')
                     break
                 if self.opts.get('caller'):
-                    print('Minion failed to authenticate with the master, '
-                          'has the minion key been accepted?')
-                    sys.exit(2)
+                    # We have a list of masters, so we should break
+                    # and try the next one in the list.
+                    if self.opts.get('local_masters', None):
```
Comment (cachedout) [inline salt/crypt.py]: This is nit-picky but this is such a sensitive area that I would feel better if we checked explicitly against a list type.

Comment (cachedout) [reply salt/crypt.py]: Oh! Oh! My fault. You're totally right.

**Comment by mattp-** [comment]
should this logic also be applied at https://github.com/garethgreenaway/salt/blob/fab25af1a975e0dd6a44ad28c52c6a3e90f9fa62/salt/crypt.py#L1099 ?

**Pull request merged**